core_types/Cargo.toml

```diff
@@ -24,11 +24,10 @@ rust-base58 = "0.0.4"
 snowflake = "1.2"
 objekt="0.1.1"
 holochain_core_types_derive = { path = "../core_types_derive" }
+holochain_sodium = { path = "../sodium" }
```

Comment:
After resolving those earlier build errors, this PR now fails because `holochain_sodium` can't compile to WASM. I'm not sure if that is fundamental issue or something that could be fixed easily.

But it seems this dependency here is only needed because of the `impl From<SodiumError> for HolochainError`.

Could we move that impl to the sodium crate? `holochain_core_types` needs to be compiled to WASM since it's used in DNAs..

Reply:
@neonphog, @zo-el.

Reply:
yep, we should pull this out. the light client web sodium @zo-el is working on now will need to be injected as a dependency once we get to the point of having core built as a wasm module